### PR TITLE
Google Cloud Tasks Sensor for queue being empty

### PR DESCRIPTION
**Pull request by vinitpayal**
This pull request adds a new sensor for Google Cloud Tasks to sense cloud task queue being empty

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Needs doc update, provider update, examples.

**Comment by vinit-tribes** [comment]
Sorry wasn't aware of the process for adding documentation & added it now. 

@mik-laj @potiuk let me know if made any mistakes in documenting addition as the format was a bit new to me. Otherwise, I think the PR is ready to be merged now.

**Comment by vinitpayal** [comment]
@potiuk please let me know if need any other help in getting it merged?

File: airflow/providers/google/cloud/example_dags/example_cloud_task.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+QUEUE_NAME = os.environ.get('GCP_QUEUE_NAME', 'testqueue')
+
+
+with models.DAG(
+    'example_gcp_cloud_tasks_sensor',
+    start_date=datetime(2022, 8, 8),
+    catchup=False,
+    tags=['example'],
+) as dag:
+    # start = EmptyOperator(task_id="start")
```
Comment (pankajastro) [inline airflow/providers/google/cloud/example_dags/example_cloud_task.py]:
```suggestion
```

File: airflow/providers/google/cloud/example_dags/example_cloud_task.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    # [START cloud_tasks_empty_sensor]
+    gcp_cloud_tasks_sensor = TaskQueueEmptySensor(
+        project_id=GCP_PROJECT_ID,
+        location=GCP_ZONE,
+        task_id='gcp_sense_cloud_tasks_empty',
+        queue_name=QUEUE_NAME,
+    )
+    # [END cloud_tasks_empty_sensor]
+    # end = EmptyOperator(task_id="end")
```
Comment (pankajastro) [inline airflow/providers/google/cloud/example_dags/example_cloud_task.py]:
```suggestion
```

File: airflow/providers/google/cloud/sensors/tasks.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+from airflow.providers.google.cloud.hooks.tasks import CloudTasksHook
+from airflow.sensors.base import BaseSensorOperator
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class TaskQueueEmptySensor(BaseSensorOperator):
+    """Pulls tasks count from a cloud task queue.
```
Comment (pankajastro) [inline airflow/providers/google/cloud/sensors/tasks.py]:
```suggestion
    """
    Pulls tasks count from a cloud task queue.
```

File: airflow/providers/google/cloud/sensors/tasks.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    ) -> None:
+        super().__init__(**kwargs)
+        self.location = location
+        self.project_id = project_id
+        self.queue_name = queue_name
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+
+    def poke(self, context: "Context") -> bool:
+        # self.log.info('Sensor checks existence of tasks in, %s', self.queue_name)
```
Comment (pankajastro) [inline airflow/providers/google/cloud/sensors/tasks.py]:
```suggestion
```

File: airflow/providers/google/cloud/sensors/tasks.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+        hook = CloudTasksHook(
+            gcp_conn_id=self.gcp_conn_id,
+            impersonation_chain=self.impersonation_chain,
+        )
+
+        # TODO uncomment page_size once https://issuetracker.google.com/issues/155978649?pli=1 gets fixed
```
Comment (pankajastro) [inline airflow/providers/google/cloud/sensors/tasks.py:78]:
```suggestion
```

**Comment by vinitpayal** [comment]
Thanks @pankajastro, suggestions well noted and applied.

**Review by potiuk — APPROVED** [review]

**Pull request merged**